src/platforms/hosted/serial_unix.c

```diff
@@ -43,6 +47,73 @@ static uint8_t read_buffer[READ_BUFFER_LENGTH];
 static size_t read_buffer_fullness = 0U;
 static size_t read_buffer_offset = 0U;
 
+/* Socket code taken from https://beej.us/guide/bgnet/ */
+static bool try_opening_network_device(const char *const name)
+{
+	// Maximum legal length of a hostname
+	char hostname[256];
+
+	// The service name or port number
+	char *service_name;
+
+	struct addrinfo addr_hints = {
+		.ai_family = AF_UNSPEC,
+		.ai_socktype = SOCK_STREAM,
+		.ai_protocol = IPPROTO_TCP,
+		.ai_flags = AI_ADDRCONFIG | AI_V4MAPPED,
+	};
+	struct addrinfo *server_info;
+
+	if (!name) {
+		return false;
+	}
+
+	// Copy the hostname to an internal array. We need to modify it
+	// to separate the hostname from the service name.
+	strncpy(hostname, name, sizeof(hostname) - 1U);
+
+	service_name = strstr(hostname, ":");
+	if (service_name == NULL) {
+		return false;
+	}
+
+	// Separate the service name / port number from the hostname
+	*service_name = '\0';
+	service_name += 1;
+
+	if (*service_name == '\0') {
+		return false;
+	}
+
+	if (getaddrinfo(hostname, service_name, &addr_hints, &server_info) != 0) {
+		return false;
+	}
+
+	// Loop through all the results and connect to the first we can.
+	struct addrinfo *p;
+	for (p = server_info; p != NULL; p = p->ai_next) {
```

Comment:
So we don't wind up introducing an unexplained single letter variable name, we're going to suggest that `server_info` be renamed to `results` and `p` to `address` or `server_addr` or similar.